doc/modules/mod_event_pusher_push.md

```diff
@@ -1,163 +1,122 @@
+* Ideal for migrations to PubSub-less deployments.
 
-This is the most versatile setup. Allows the clients to use all push notifications configurations:
+#### Drawbacks
+* More complex configuration on the server side
+* Pays the PubSub performance penalty when the PubSub path is taken
 
-* Push service handled by another cluster via XEP-0357
-* Push service handled locally via XEP-0357
-* PubSub-less
 
-```Erlang
-{mod_pubsub, [{plugins, [<<"push">>]}]}, % mandatory minimal config
+## Plugin module
 
-{mod_event_pusher, [
-    {backends, [
-        {push, [
-            {backend, mnesia}, % optional
-            {wpool, [{workers, 200}]}, % optional
-            {plugin_module, mod_event_pusher_push_plugin_defaults}, % optional
-            {virtual_pubsub_hosts, ["virtual.@HOST@"]}
-        ]}
-    ]}
-]}
-```
+You can also control the format of the "sender" of the push notification (which ultimately becomes
+the title of push notification) and filter which messages will trigger the notification.
+In order to achieve that, you need to create a plugin module that implements the
+`mod_event_pusher_push_plugin` behaviour and enable this plugin in the `plugin_module` section as
+above.
 
-### Migration from XEP-0357 to virtual hosts
+A plugin module handles the dynamic configuration of push notifications. 
+It contains the filtering and custom logic for notifying about messages.
 
-This is an example of how you can migrate the existing setup to the new model.
-PubSub service still exists (because you might want to reuse it e.g. for publishing E2E encryption keys).
-However, its domain is overridden for the purpose of sending push notifications.
-Please note the value of `virtual_pubsub_hosts` option.
-`pubsub.@HOST@` is the default domain for `mod_pubsub`.
+Two plugin implementations are provided.
+They offer different behaviour considering unacknowledged messages when using [Stream Management][XEP-0198]:
 
-```Erlang
-{mod_pubsub, [{plugins, [<<"push">>]}]}, % mandatory minimal config
 
-{mod_event_pusher, [
-    {backends, [
-        {push, [
-            {backend, mnesia}, % optional
-            {wpool, [{workers, 200}]}, % optional
-            {plugin_module, mod_event_pusher_push_plugin_defaults}, % optional
-            {virtual_pubsub_hosts, ["pubsub.@HOST@"]}
-        ]}
-    ]}
-]}
-```
-
-## Plugin module
+* `mod_event_pusher_push_plugin_defaults`, which implements an older behaviour. It does not notify
+  the user of unacknowledged messages immediately after detecting a lost connection to the user.
+* `mod_event_pusher_push_plugin_enhanced`, which pushes notifications as soon as the server detects
+  that the client has disconnected and waits for stream resumption (by an `unack_msg_event` event
+  being generated through the `unacknowledged_message` hook). The immediate notification prevents
+  the unneeded suspension of the client's application, if there are any not yet unacknowledged
```

Comment:
any not yet unacknowledged messages? something is wrong here.

Reply:
Damn copy&pasting 😂